packages/core/src/routes/callback.ts

```diff
@@ -4,21 +4,37 @@
  */
 
 import type Koa from 'koa';
-import koaBody from 'koa-body';
+import { koaBody } from 'koa-body';
 import Router from 'koa-router';
 
 import RequestError from '#src/errors/RequestError/index.js';
 import assertThat from '#src/utils/assert-that.js';
 
+// Edited from https://stackoverflow.com/a/74743075/12514940
+function isStringRecord(object: unknown): object is Record<string, string> {
+  if (typeof object !== 'object' || object === null) {
+    return false;
+  }
+
+  if (Array.isArray(object)) {
+    return false;
+  }
+
+  if (Object.getOwnPropertySymbols(object).length > 0) {
+    return false;
+  }
+
+  return Object.getOwnPropertyNames(object).every(
+    // @ts-expect-error This is a type guard
+    (property) => typeof object[property] === 'string'
+  );
+}
+
 function callbackRoutes<T extends Router>(router: T) {
   router.post('/callback/:connectorId', koaBody(), async (ctx) => {
-    assertThat(
-      typeof ctx.request.body === 'object' && ctx.request.body !== null,
-      new RequestError('oidc.invalid_request')
-    );
+    assertThat(isStringRecord(ctx.request.body), new RequestError('oidc.invalid_request'));
```

Comment:
Just curious: why do we need to write a type guard ourselves instead of using zod?

Reply:
i forgot this